gtp_cmd_handlers.py

```diff
@@ -0,0 +1,340 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from datetime import datetime, timedelta
+import sys
+import itertools
+import json
+import sgf_wrapper
+import go
+import coords
+
+
+def dbg(fmt, *args):
+    print(fmt % args, file=sys.stderr, flush=True)
+
+
+def translate_gtp_color(gtp_color):
+    if gtp_color.lower() in ["b", "black"]:
+        return go.BLACK
+    if gtp_color.lower() in ["w", "white"]:
+        return go.WHITE
+    raise ValueError("invalid color {}".format(gtp_color))
+
+
+class BasicCmdHandler(object):
+    """GTP command handler for basic play commands."""
+
+    def __init__(self, player, courtesy_pass=False):
+        self._komi = 6.5
+        self._player = player
+        self._player.initialize_game()
+        self._courtesy_pass = courtesy_pass
+
+    def cmd_boardsize(self, n: int):
+        if n != go.N:
+            raise ValueError("unacceptable size")
+
+    def cmd_clear_board(self):
+        position = self._player.get_position()
+        if (self._player.get_result_string()
+                and position and len(position.recent) > 1):
+            try:
+                sgf = self._player.to_sgf()
+                with open(datetime.datetime.now().strftime("%Y-%m-%d-%H:%M.sgf"), 'w') as f:
+                    f.write(sgf)
+            except NotImplementedError:
+                pass
+            except:
+                print("Error saving sgf", file=sys.stderr, flush=True)
+        self._player.initialize_game(go.Position(komi=self._komi))
+
+    def cmd_komi(self, komi: float):
+        self._komi = komi
+        self._player.get_position().komi = komi
+
+    def cmd_play(self, arg0: str, arg1=None):
```

Comment:
did you mean to leave this as arg0/arg1?

Reply:
Yes, unfortunately.

If the two argument variant of `play` was `play move color` instead of `play color move`, I could just make `color` optional. But as it is, I don't see a way of sensibly naming the parameters. I guess I could have gone with `color_or_move` and `maybe_move` or something, but I don't think that's any more readable than just `arg0` and `arg1`.